homeassistant/components/aurora/binary_sensor.py

```diff
@@ -1,146 +1,77 @@
-import requests
-import voluptuous as vol
-
-from homeassistant.components.binary_sensor import PLATFORM_SCHEMA, BinarySensorEntity
-from homeassistant.const import ATTR_ATTRIBUTION, CONF_NAME
-import homeassistant.helpers.config_validation as cv
-from homeassistant.util import Throttle
-
-_LOGGER = logging.getLogger(__name__)
-
-ATTRIBUTION = "Data provided by the National Oceanic and Atmospheric Administration"
-CONF_THRESHOLD = "forecast_threshold"
-
-DEFAULT_DEVICE_CLASS = "visible"
-DEFAULT_NAME = "Aurora Visibility"
-DEFAULT_THRESHOLD = 75
-
-HA_USER_AGENT = "Home Assistant Aurora Tracker v.0.1.0"
-
-MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=5)
-
-URL = "http://services.swpc.noaa.gov/text/aurora-nowcast-map.txt"
-
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {
-        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
-        vol.Optional(CONF_THRESHOLD, default=DEFAULT_THRESHOLD): cv.positive_int,
-    }
+from homeassistant.components.binary_sensor import BinarySensorEntity
+from homeassistant.const import ATTR_NAME
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import AuroraDataUpdateCoordinator
+from .const import (
+    ATTR_IDENTIFIERS,
+    ATTR_MANUFACTURER,
+    ATTR_MODEL,
+    ATTRIBUTION,
+    COORDINATOR,
+    DOMAIN,
 )
 
+_LOGGER = logging.getLogger(__name__)
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
-    """Set up the aurora sensor."""
-    if None in (hass.config.latitude, hass.config.longitude):
-        _LOGGER.error("Lat. or long. not set in Home Assistant config")
-        return False
 
-    name = config[CONF_NAME]
-    threshold = config[CONF_THRESHOLD]
+async def async_setup_entry(hass, entry, async_add_entries):
+    """Set up the binary_sensor platform."""
+    coordinator = hass.data[DOMAIN][entry.entry_id][COORDINATOR]
+    name = coordinator.name
 
-    try:
-        aurora_data = AuroraData(hass.config.latitude, hass.config.longitude, threshold)
-        aurora_data.update()
-    except requests.exceptions.HTTPError as error:
-        _LOGGER.error("Connection to aurora forecast service failed: %s", error)
-        return False
+    entity = AuroraSensor(coordinator, name)
 
-    add_entities([AuroraSensor(aurora_data, name)], True)
+    async_add_entries([entity])
 
 
-class AuroraSensor(BinarySensorEntity):
+class AuroraSensor(CoordinatorEntity, BinarySensorEntity):
     """Implementation of an aurora sensor."""
 
-    def __init__(self, aurora_data, name):
-        """Initialize the sensor."""
-        self.aurora_data = aurora_data
+    def __init__(self, coordinator: AuroraDataUpdateCoordinator, name):
+        """Define the binary sensor for the Aurora integration."""
+        super().__init__(coordinator=coordinator)
+
         self._name = name
+        self.coordinator = coordinator
+        self._unique_id = (
+            f"{str(self.coordinator.latitude)}_{str(self.coordinator.longitude)}"
```

Comment:
No need to cast to string, happens automatically.
```suggestion
            f"{self.coordinator.latitude}_{self.coordinator.longitude}"
```